internal/types/document.go

```diff
@@ -387,6 +363,35 @@ func (d *Document) isKeyDuplicate(targetKey string) bool {
 	return false
 }
 
+// MoveIDToTheFirstIndex sets the _id field of the document at the first position.
+// If the _id field is not present, it does nothing.
+func (d *Document) MoveIDToTheFirstIndex() {
+	if !d.Has("_id") {
+		return
+	}
+
+	idIdx := 0
+
+	if d.fields[idIdx].key == "_id" {
+		return
+	}
+
+	for i, key := range d.Keys() {
+		if key == "_id" {
+			idIdx = i
+			break
+		}
+	}
+
+	if idIdx == 0 {
+		return
```

Comment:
This place is reached if `MoveIDToTheFirstIndex` is called for a document that doesn't contain `_id` at all. Is it a possible (correct) case, or should we panic if that happened?

Reply:
That check should be removed as we have it before the for loop.